target/linux/mediatek/image/mt7623.mk

```diff
@@ -1,11 +1,39 @@
 KERNEL_LOADADDR := 0x80008000
 
+BOOTFS_BLOCK_SIZE := 1024
+BOOTFS_BLOCKS := $(shell echo $$(($(CONFIG_BANANA_PI_BOOT_PARTSIZE)*1024*1024/$(BOOTFS_BLOCK_SIZE))))
+
+define Build/banana-pi-sdcard
+	rm -f $@.boot
+	mkfs.fat -C $@.boot $(BOOTFS_BLOCKS)
+
+	./gen_banana_pi_img.sh emmc $@.emmc \
+		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin
+
+	mcopy -i $@.boot $(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-uEnv.txt ::uEnv.txt
+	mcopy -i $@.boot $(IMAGE_KERNEL) ::uImage
+	mcopy -i $@.boot $@.emmc ::eMMCboot.bin
+	./gen_banana_pi_img.sh sd $@ \
+		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-preloader.bin \
+		$(STAGING_DIR_IMAGE)/$(UBOOT_TARGET)-u-boot.bin \
+		$@.boot \
+		$(IMAGE_ROOTFS) \
+		$(CONFIG_BANANA_PI_BOOT_PARTSIZE) \
+		$(CONFIG_TARGET_ROOTFS_PARTSIZE)
+endef
+
 define Device/bpi_bananapi-r2
   DEVICE_VENDOR := Bpi
   DEVICE_MODEL := Banana Pi R2
   DEVICE_DTS := mt7623n-bananapi-bpi-r2
   SUPPORTED_DEVICES := bananapi,bpi-r2
-  DEVICE_PACKAGES := kmod-usb-ohci kmod-usb2 kmod-usb3 kmod-ata-ahci-mtk
+  DEVICE_PACKAGES := kmod-usb-ohci kmod-usb2 kmod-usb3 kmod-ata-ahci-mtk \
+                     mt7623n_preloader u-boot-mt7623n_bpir2 mkf2fs
+  UBOOT_TARGET := mt7623n_bpir2
+  IMAGES += sdcard.bin.gz
```

Comment:
Does it still use sysupgrade.bin? Does the sysupgrade image still boot well?

Reply:
We're just working on forward porting the sysupgrade support.

Reply:
No, I've removed sysupgrade.bin and we upgrade using the same SD image.